lib/ldap_fluff/netiq_member_service.rb

```diff
@@ -0,0 +1,81 @@
+require 'net/ldap'
+
+# handles the naughty bits of posix ldap
+class LdapFluff::NetIQ::MemberService < LdapFluff::GenericMemberService
+
+  def initialize(ldap, config)
+    @attr_login = (config.attr_login || 'memberuid')
+    super
+  end
+
+  def find_user(uid, base_dn = @base)
+    user = @ldap.search(:filter => name_filter(uid), :base => base_dn)
+    raise UIDNotFoundException if (user.nil? || user.empty?)
+    user
+  end
+
+  def find_by_dn(search_dn)
+    entry, base = search_dn.split(/(?<!\\),/, 2)
+    _entry_attr, entry_value = entry.split('=', 2)
+    entry_value = entry_value.gsub('\,', ',')
+    user = @ldap.search(:filter => name_filter(entry_value, "workforceid"), :base => base)
+    raise self.class::UIDNotFoundException if (user.nil? || user.empty?)
+    user
+  end
+
+  def get_logins(userlist)
+    userlist.map(&:downcase!)
```

Comment:
Is it a good idea to use an in place replacement? `downcase!` feels insecure. How about
```suggestion
    userlist = userlist.map(&:downcase)
```

Or combine it in the method below with `find_by_dn(current_user.downcase)`.